canonical-web-and-design/maas-ui
# refactor(ui): convert DoubleRow and TableMenu to typescript

## Done

- Convert DoubleRow and TableMenu to typescript.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- View the machine list.
- The double rows should appear correctly and the menus should open on the rows.

## Fixes

Part of canonical-web-and-design/maas-squad/issues/2284.
